R/util.R

```diff
@@ -705,3 +705,26 @@ pivot_ <- function(data, formula, value_col = NULL, fun.aggregate = mean, fill =
     as.data.frame %>%
     tibble::rownames_to_column(var = cname)
 }
+
+#'convert values to binary label
+binary_label <- function(val) {
+  if(is.factor(val)){
+    # Need to subtract 1 because the first level in factor is regarded as 1
+    # though it should be FALSE.
+    val <- as.logical(as.integer(val) - 1)
+  } else {
+    logi_val <- as.logical(val)
+    # if the values are non-zero values,
+    # larger value should be regarrded as FALSE
+    if(all(logi_val[!is.na(logi_val)])){
+      unique_val <- unique(val[!is.na(val)])
+      if(length(unique_val) == 2) {
+        logi_val <- val == max(unique_val)
+      } else if (length(unique_val) > 2) {
+        stop("actual labels can't have more than 2 unique values")
```

Comment:
To be a generic utility function, we should not assume that the column is called "actual".